edsnlp/pipes/qualifiers/family/family.py

```diff
@@ -187,7 +187,8 @@ def set_extensions(self) -> None:
         if not Doc.has_extension("family"):
             Doc.set_extension("family", default=[])
 
-    def process(self, doc: Doc) -> FamilyResults:
+    def process(self, doc_like: Union[Doc, Span]) -> FamilyResults:
+        doc = self.ensure_doc(doc_like)
         matches = self.get_matches(doc)
```

Comment:
@Thomzoy Isn't this misleading, wrt the Doc|Span parameter type ? This function claims it can process a Span, so users might assume it only processes entities inside the span, when in reality it performs the matching on the full document, regardless of its doc_like parameter

Reply:
Ah yes, I mixed `span.as_doc()` vs `span.doc`.
I think this poses another problem : `span.as_doc()` creates a new document, detached from the original doc, so any new attribute added won't be saved. If this feature actually used ? If not, I'd rather remove it at the moment, as this changes EDS-NLP API and makes it non uniform across our pipes (somes pipes accept both Span and Doc, some only Doc)

Reply:
Yes it was the exact goal : *Normal* use of qualifiers will still be through the `__call__` method, but if you want to qualify part of a document (for instance, an `assign` span from the ContextualMatcher) without *polluting* the initial document, you'll use `process`.

And in the end the `__call__` method stays consistent (only accepting Doc)